### DISCUSSION TITLE
where does one get a hardware-configuration.nix file

### DISCUSSION THREAD

**Discussion by GregHilston** [nix-community/NixOS-WSL#605] [Q&A] [answered]
Is it still correct to leverage 

`$ nixos-generate-config`?

I ask, as the output of this file failed to work as is and required manual editing. One of the drives has an empty string for a value.

**Comment by SuperSandro2000** [comment]
No, you don't.

**Reply by GregHilston** [reply]
@SuperSandro2000 okay understood. Any advice on how to proceed then? I ask, as I haven't seen it discussed in any writing or mentioned in any video.

I'm more than happy to submit a PR after the fact, adding in more documention

**Reply by SuperSandro2000** [reply, marked as answer]
Proceed with what? You should be good to go by just removing the reference to hardware-configuration and them load the nixos-wsl module like normal.

**Reply by GregHilston** [reply]
@SuperSandro2000 My apologies, I *completely* misunderstood what the `NixOS-WSL` repository was doing. I did not relaize that it was a module, that provided value for working on WSL.  I thought this project was simply getting NixOS to run on WSL, and nothing more. IE getting the actual OS setup, and not providing a handy module for configuring your NixOS to run on WSL.

If I understand correctly, I basically can take [my NixOS setup](https://github.com/GregHilston/toolbox/tree/master/nixos), and add a new entry in my [flake.nix](https://github.com/GregHilston/toolbox/blob/master/nixos/flake.nix#L37), and a corresponding entry in my [hosts/pcs/ subidrectory](https://github.com/GregHilston/toolbox/tree/master/nixos/hosts/pcs), and use the `NixOS-WSL` module as described in [the official documentation "nix-flakes"](https://nix-community.github.io/NixOS-WSL/how-to/nix-flakes.html). This should help me a ton with configuring NixOS on WSL.

And I don't need a `hardware-configuration.nix` file at all. Being new to Nix, I thought that this file was a requirement for using NixOS.

I'll close this discussion, marking it as answered, and ask a more specific question if I have any in the future. Thank you @SuperSandro2000
